File: backend/src/main/java/com/lawencon/ticketms/service/TicketService.java
package com.lawencon.ticketms.service;

import com.lawencon.ticketms.dto.ticket.TicketInsertReqDto;
import com.lawencon.ticketms.dto.ticket.TicketInsertResDto;
import com.lawencon.ticketms.dto.ticket.TicketListResDto;
import com.lawencon.ticketms.dto.ticket.TicketResDto;
import com.lawencon.ticketms.dto.ticket.TicketUpdateReqDto;
import com.lawencon.ticketms.dto.ticket.TicketUpdateResDto;

public interface TicketService {
	TicketInsertResDto insert(TicketInsertReqDto data);
	TicketResDto getById(Long id);
	TicketListResDto getAll();
	
	TicketListResDto getAllIdCust(Long id);
	TicketListResDto getAllIdPic(Long id);
	TicketUpdateResDto updateStatus(TicketUpdateReqDto data);
}

File: frontend/src/app/service/ticket.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";
import { BASE_URL } from "../constant/base.url";
import { TicketInsertReqDto } from "../dto/ticket/ticket.insert.req.dto";
import { TicketListResDto } from "../dto/ticket/ticket.list.res.dto";
import { TicketResDto } from "../dto/ticket/ticket.res.dto";
import { TicketUpdateReqDto } from "../dto/ticket/ticket.update.req.dto";

@Injectable({
    providedIn : 'root'
})
export class TicketService {

    constructor(private http : HttpClient) {}

    getAllIdCust(id : number) : Observable<TicketListResDto> {
        return this.http.get<TicketListResDto>(`${BASE_URL.LOCALHOST}/tickets/customer/${id}`)
    }

    insert(data :TicketInsertReqDto) : Observable<TicketInsertReqDto>{
        return this.http.post<TicketInsertReqDto>(`${BASE_URL.LOCALHOST}/tickets`, data);
    }

    getAllIdPic(id : number) : Observable<TicketListResDto> {
        return this.http.get<TicketListResDto>(`${BASE_URL.LOCALHOST}/tickets/pic/${id}`)
    }

    getById(id:number) : Observable<TicketResDto>{
        return this.http.get<TicketResDto>(`${BASE_URL.LOCALHOST}/tickets/${id}`)
    }
    
    update(data :TicketUpdateReqDto) : Observable<TicketUpdateReqDto>{
        return this.http.put<TicketUpdateReqDto>(`${BASE_URL.LOCALHOST}/tickets`, data);
    }

}